pkg/stub/handler.go

```diff
@@ -559,38 +552,47 @@ func (h *Handler) processFiles(dir string, files []os.FileInfo, opcfg *v1alpha1.
 	return nil
 }
 
+func (h *Handler) coreUpdateDockerPullSpec(oldreg, newreg string, oldies []string) string {
+	// see if the imagestream on file (i.e. the openshift/library content) is
+	// of the form "reg/repo/img" or "repo/img"
+	hasRegistry := false
+	if strings.Count(oldreg, "/") == 2 {
+		hasRegistry = true
+	}
+	replaced := false
+	for _, old := range oldies {
+		if strings.HasPrefix(oldreg, old) {
+			oldreg = strings.Replace(oldreg, old, newreg, 1)
+			replaced = true
+			break
+		}
+	}
+	if !replaced {
+		if !hasRegistry {
+			oldreg = newreg + "/" + oldreg
+		} else {
+			// the content from openshift/library has something odd in in ... replace the registry piece
+			parts := strings.Split(oldreg, "/")
+			oldreg = newreg + "/" + parts[1] + "/" + parts[2]
+		}
+	}
```

Comment:
I would do this logic slightly differently. I think it's a mistake to call strings.Replace() on a value that we previously concluded does not have a registry prefix.  (it's unlikely to replace anything, but it could).

Should probably do:

```
if hasRegistry {
   // do search+replace
} else {
  // prepend the value w/ the replacement registry value
}
```

Reply:
part of next push ... chose to still include the openshift/library content hasRegistry, but not one of our presumed constants, piece